### PR TITLE
T32: Implement load/store single (immediate)

### PR DESCRIPTION
**Pull request by merryhime**
Minor progress on #2668.

### CODE REVIEW

**Comment by gdkchan** [comment]
Do you plan to include tests on this one too or it's ready for review?

**Comment by merryhime** [comment]
Thinking about how I should best implement tests.

**Comment by merryhime** [comment]
Tests added (had to spend some time on test generation stuff).

File: Ryujinx.Tests/Cpu/CpuTest32.cs [hunk 1]
```diff
@@ -286,6 +286,35 @@ public void RunPrecomputedTestCase(PrecomputedThumbTestCase test)
             Assert.That(GetContext().Pstate, Is.EqualTo(finalCpsr));
         }
 
+        public void RunPrecomputedTestCase(PrecomputedMemoryThumbTestCase test)
+        {
+            byte[] testmem = new byte[Size];
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTest32.cs:291]: nit: `testMem`.

File: Ryujinx.Tests/Cpu/PrecomputedMemoryThumbTestCase.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using System;
+
+namespace Ryujinx.Tests.Cpu
+{
+    public struct PrecomputedMemoryThumbTestCase
+    {
+        public ushort[] Instructions;
+        public uint[] StartRegs;
+        public uint[] FinalRegs;
+        public Tuple<ulong, ushort>[] MemoryDelta;
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/PrecomputedMemoryThumbTestCase.cs:10]: you could use named tuples here. Like `public (ulong X, ushort Y) MemoryDelta;` (replacing X and Y here with actual names).

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Review by AcK77 — APPROVED** [review]

**Pull request merged**